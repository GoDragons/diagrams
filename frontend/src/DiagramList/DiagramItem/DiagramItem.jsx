import React from "react";
import { Button, Space, Typography, Row, Col } from "antd";
import Card from "Card/Card";
import { Link } from "react-router-dom";

import AvatarList from "AvatarList/AvatarList";

import "./DiagramItem.scss";

export default function DiagramItem({
  diagramId,
  latestVersionId,
  diagramName,
  versions,
  participants,
}) {
  let lastModifiedTimestamp;
  let createdTimestamp;

  if (versions) {
    lastModifiedTimestamp = versions[0].lastModified;
    createdTimestamp = parseInt(versions[versions.length - 1].versionId);
  }

  return (
    <Col span={24} className="diagram-item">
      <Card>
        <Link
          to={`/diagrams/${diagramId}/${latestVersionId}/edit`}
          className="diagram-name"
        >
          {diagramName}
        </Link>
        <Typography.Paragraph className="created">
          Created {window.moment(createdTimestamp).fromNow()}
        </Typography.Paragraph>
        <Typography.Paragraph className="last-modified">
          Last modified {window.moment(lastModifiedTimestamp).fromNow()}
        </Typography.Paragraph>

        <Row>
          <Col span={12}>
            <Space>
              <Link to={`/diagrams/${diagramId}/${latestVersionId}/edit`}>
                <Button type="primary">Edit</Button>
              </Link>
              <Link to={`/diagrams/${diagramId}/details`}>
                <Button type="secondary">Details</Button>
              </Link>
            </Space>
          </Col>
          <Col span={12}>
            <AvatarList users={participants} />
          </Col>
        </Row>
      </Card>
    </Col>
  );
}
