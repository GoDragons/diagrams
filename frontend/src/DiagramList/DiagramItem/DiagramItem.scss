@import "../../Variables";

.diagram-item {
  &.ant-col-6 {
    max-width: 25rem;
  }

  .card {
    .created {
      margin-bottom: 0;
    }

    .diagram-name {
      margin: 0;
      font-size: 1rem;
      color: $color1;
    }
  }
}
