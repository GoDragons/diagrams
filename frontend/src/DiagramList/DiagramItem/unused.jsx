// function deleteVersion(versionId) {
//   axios
//     .post(
//       `${REST_API_URL}/delete-version`,
//       { diagramId, versionId },
//       {
//         headers: {
//           Authorization: userCredentials.accessToken.jwtToken,
//         },
//       }
//     )
//     .then(refreshList)
//     .catch((e) => console.log(`Could not delete version:`, e.response.data));
// }
// function deleteDiagram() {
//   axios
//     .post(
//       `${REST_API_URL}/delete-diagram`,
//       { diagramId },
//       {
//         headers: {
//           Authorization: userCredentials.accessToken.jwtToken,
//         },
//       }
//     )
//     .then(refreshList)
//     .catch((e) => console.log(`Could not delete diagram:`, e.response.data));
// }

// function displayVersions(versions) {
//   if (!versions || versions.length === 1) {
//     return null;
//   }
//   const versionElements = versions.slice(1).map((version) => {
//     const { versionId, versionName, lastModified } = version;
//     return (
//       <li key={`${diagramId}-${versionId}`} className="version-item">
//         <Link to={`/diagrams/${diagramId}/${versionId}`}>
//           <span className="version-name">{versionName}</span>
//           <span className="last-modified">
//             {window.moment(lastModified).format("DD MMM YYYY - HH:mm:ss")}
//           </span>
//         </Link>
//         <button onClick={(e) => deleteVersion(versionId)}>
//           Delete version
//         </button>
//       </li>
//     );
//   });

//   return (
//     <div className="version-list-container">
//       <p className="version-id">Versions: </p>
//       <ul className="versions">{versionElements}</ul>
//     </div>
//   );
// }

{
  /* <button onClick={deleteDiagram}>Delete diagram</button> */
}

{
  /* <Link to={`/diagrams/${diagramId}/${latestVersionId}`}>
        <h3 className="title">{diagramName}</h3>
      </Link> */
}
