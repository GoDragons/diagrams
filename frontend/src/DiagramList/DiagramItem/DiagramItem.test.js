import React from "react";
import { shallow } from "enzyme";
import { Link } from "react-router-dom";
import DiagramItem from "./DiagramItem";

describe("DiagramItem component", () => {
  beforeAll(() => {
    global.moment = jest.fn().mockImplementation((timestamp) => {
      return {
        fromNow: () => timestamp + "_ago",
      };
    });
  });
  xit("renders without crashing", () => {
    shallow(<DiagramItem />);
  });
  it("contains the correct link to the 'edit' page in the diagram name", () => {
    const component = shallow(
      <DiagramItem
        diagramId="1234"
        latestVersionId="abcd"
        diagramName="Test diagram"
      />
    );
    const target = component.find(".diagram-name");
    expect(target).toMatchSnapshot();
  });
  it("contains 'created'", () => {
    const component = shallow(
      <DiagramItem versions={[{ versionId: "56789" }]} />
    );
    const target = component.find(".created");
    expect(target).toMatchSnapshot();
  });

  it("contains 'last modified'", () => {
    const component = shallow(
      <DiagramItem versions={[{ lastModified: "9876" }]} />
    );
    const target = component.find(".last-modified");
    expect(target).toMatchSnapshot();
  });

  // it("links to the correct page with the 'edit' button", )
});
