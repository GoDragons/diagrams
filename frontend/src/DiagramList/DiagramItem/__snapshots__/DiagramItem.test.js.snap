// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiagramItem component contains 'created' 1`] = `
<Paragraph
  className="created"
>
  Created 
  56789_ago
</Paragraph>
`;

exports[`DiagramItem component contains 'last modified' 1`] = `
<Paragraph
  className="last-modified"
>
  Last modified 
  9876_ago
</Paragraph>
`;

exports[`DiagramItem component contains the correct link to the 'edit' page in the diagram name 1`] = `
<Link
  className="diagram-name"
  to="/diagrams/1234/abcd/edit"
>
  Test diagram
</Link>
`;
